### DISCUSSION TITLE
The total charge of the smiles string is not being computed correctly

### DISCUSSION THREAD

**Discussion by xiki-tempula** [rdkit/rdkit#4334] [Q&A] [answered]
I'm interested in using rdkit to obtain the total charge of the smiles string for my QM calculations.
My pipeline is that I will add hydrogen atoms, calculate the Gasteiger Charges and the sum of Gasteiger Charges is the total charge of the molecules.

```
mol = Chem.MolFromSmiles(string)
mol_H = Chem.AddHs(mol)
AllChem.ComputeGasteigerCharges(mol_H)
contribs = [mol_H.GetAtomWithIdx(i).GetDoubleProp('_GasteigerCharge')
                    for i in range(mol_H.GetNumAtoms())]
charge = int(np.round(sum(contribs)))
```

However, this pipeline won't work for the string 'Cc1ccc(cc1)Cc2c[n-]c(nc2=[N+]3CCOCC3)[NH-]', which clearly has total charge of -1, but the above pipeline will give a total charge of 0, `sum(contribs)` gives -0.3333333.

Similarly, 'Cc1cc(cc(c1O)C(C)(C)C)Cc2c[nH+]c(nc2[NH-])[NH-]' gives `sum(contribs) == -0.5`.

the rdkit version is '2021.03.4'.

BTW, is there other recommended way of computing the total charge of the molecule based on the smiles string? Thank you.

**Comment by ptosco** [comment, marked as answer]
@xiki-tempula I would just compute the sum of formal charges:
```
sum(a.GetFormalCharge() for a in mol.GetAtoms())
```
